climada/trajectories/interpolation.py

```diff
@@ -0,0 +1,141 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+This modules implements different sparce matrices interpolation approaches.
+
+"""
+
+import logging
+from abc import ABC, abstractmethod
+
+import numpy as np
+from scipy.sparse import csr_matrix, lil_matrix
+
+LOGGER = logging.getLogger(__name__)
+
+
+class InterpolationStrategy(ABC):
+    """Interface for interpolation strategies."""
+
+    @abstractmethod
+    def interpolate(self, imp_E0, imp_E1, time_points: int) -> list: ...
+
+
+class LinearInterpolation(InterpolationStrategy):
+    """Linear interpolation strategy."""
+
+    def interpolate(self, imp_E0, imp_E1, time_points: int):
+        try:
+            return self.interpolate_imp_mat(imp_E0, imp_E1, time_points)
+        except ValueError as e:
+            if str(e) == "inconsistent shapes":
+                raise ValueError(
+                    "Interpolation between impact matrices of different shapes"
```

Comment:
Very cryptic for a user if triggered not in this method specifically, when a method calls this method.